sonic-xcvrd/scripts/xcvrd

```diff
@@ -0,0 +1,430 @@
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "xcvrd"
+XCVRD_MODULE_NAME = "xcvrd"
+
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+MINIGRAPH_PATH = '/etc/sonic/minigraph.xml'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
+
+PLATFORM_SPECIFIC_MODULE_NAME = "sfputil"
+PLATFORM_SPECIFIC_CLASS_NAME = "SfpUtil"
+
+# Global platform-specific sfputil class instance
+platform_sfputil = None
+
+# platform directory in base image
+PLATFORM_ROOT = "/usr/share/sonic/device"
+
+# platform root directory inside docker
+PLATFORM_ROOT_DOCKER = "/usr/share/sonic/platform"
+
+REDIS_HOSTNAME = "localhost"
+REDIS_PORT = 6379
+REDIS_TIMEOUT_USECS = 0
+SELECT_TIMEOUT = 1000
+
+DOM_INFO_UPDATE_PERIOD = 60
+
+SFP_STATUS_PLUGIN = '1'
+SFP_STATUS_PLUGOUT = '0'
+
+PHYSICAL_PORT_NOT_EXIST = -1
+SFP_EEPROM_NOT_READY = -2
+
+TEMP_UNIT = 'C'
+VOLT_UNIT = 'Volts'
+POWER_UNIT = 'dBm'
+BIAS_UNIT = 'mA'
+
+#========================== Syslog wrappers ==========================
+
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+#========================== Signal Handling ==========================
+
+def signal_handler(sig, frame):
+    if sig == signal.SIGHUP:
+        log_info("Caught SIGHUP - ignoring...")
+        return
+    elif sig == signal.SIGINT:
+        log_info("Caught SIGINT - exiting...")
+        sys.exit(128 + sig)
+    elif sig == signal.SIGTERM:
+        log_info("Caught SIGTERM - exiting...")
+        sys.exit(128 + sig)
+    else:
+        log_warning("Caught unhandled signal '" + sig + "'")
+
+
+#============ Functions to load platform-specific classes ============
+
+# Returns platform and HW SKU
+def get_platform_and_hwsku():
+    try:
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
+                stdout=subprocess.PIPE,
+                shell=False,
+                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        platform = stdout.rstrip('\n')
+
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-m', MINIGRAPH_PATH, '-v', HWSKU_KEY],
+                stdout=subprocess.PIPE,
+                shell=False,
+                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        hwsku = stdout.rstrip('\n')
+    except OSError, e:
```

Comment:
>except [](start = 4, length = 6)

try-except added no valued except a message which may be not accurate. Suggest remove.